src/main/java/com/esri/samples/na/service_areas/ServiceAreaTaskController.java

```diff
@@ -0,0 +1,259 @@
+
+import javafx.fxml.FXML;
+import javafx.geometry.Point2D;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Alert.AlertType;
+import javafx.scene.control.Button;
+import javafx.scene.control.ToggleButton;
+import javafx.scene.input.MouseButton;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.Polyline;
+import com.esri.arcgisruntime.geometry.PolylineBuilder;
+import com.esri.arcgisruntime.geometry.SpatialReference;
+import com.esri.arcgisruntime.geometry.SpatialReferences;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.symbology.PictureMarkerSymbol;
+import com.esri.arcgisruntime.symbology.SimpleFillSymbol;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.tasks.networkanalysis.PolylineBarrier;
+import com.esri.arcgisruntime.tasks.networkanalysis.ServiceAreaFacility;
+import com.esri.arcgisruntime.tasks.networkanalysis.ServiceAreaParameters;
+import com.esri.arcgisruntime.tasks.networkanalysis.ServiceAreaPolygon;
+import com.esri.arcgisruntime.tasks.networkanalysis.ServiceAreaResult;
+import com.esri.arcgisruntime.tasks.networkanalysis.ServiceAreaTask;
+
+public class ServiceAreaTaskController {
+
+  @FXML private MapView mapView;
+  @FXML private ToggleButton btnAddFacility;
+  @FXML private ToggleButton btnAddBarrier;
+  @FXML private Button btnShowServiceArea;
+  @FXML private Button btnReset;
+
+  // holds all location to find service areas around
+  private List<ServiceAreaFacility> serviceAreaFacilities;
+  // different colors to display multiple service areas
+  private List<SimpleFillSymbol> fillSymbols;
+
+  // task to find service area around a location
+  private ServiceAreaTask task;
+  // used to solve task above
+  private ServiceAreaParameters serviceAreaParameters;
+  // holds all facilities that are displayed to the mapview
+  private GraphicsOverlay facilityOverlay;
+  // holds all polygon service areas that are displayed to the mapview
+  private GraphicsOverlay polygonOverlay;
+  // holds all barriers that are displayed to mapview
+  private GraphicsOverlay barrierOverlay;
+  // used to make barriers
+  private PolylineBuilder barrierBuilder;
+  // used for placing geometry on mapview
+  private SpatialReference spatialReference = SpatialReferences.getWebMercator();
+
+  public void initialize() {
+
+    // create map of creates and add to mapview
+    ArcGISMap map = new ArcGISMap(Basemap.createStreets());
+    mapView.setMap(map);
+    // set mapview to San Francisco
+    mapView.setViewpoint(new Viewpoint(37.77, -122.41, 40000));
+
+    createServiceAreaTask();

```

Comment:
My general guideline is to not create methods for logic that will only ever be called once.